### PR TITLE
satellite/satellitedb: add professional user fields to db interface

### PR DESCRIPTION
**Pull request by mbouzi**
What: Not Ready For Review

Why:

Please describe the tests:
 - Test 1:
 - Test 2:
 
Please describe the performance impact:

## Code Review Checklist (to be filled out by reviewer)
 - [x] NEW: Are there any Satellite database migrations? Are they forwards _and_ backwards compatible? 
 - [x] Does the PR describe what changes are being made?
 - [x] Does the PR describe why the changes are being made?
 - [x] Does the code follow [our style guide](https://github.com/storj/docs/blob/main/code/Style.md)?
 - [x] Does the code follow [our testing guide](https://github.com/storj/docs/blob/main/code/Testing.md)?
 - [x] Is the PR appropriately sized? (If it could be broken into smaller PRs it should be)
 - [x] Does the new code have enough tests? (*every* PR should have tests or justification otherwise. Bug-fix PRs especially)
 - [x] Does the new code have enough documentation that answers "how do I use it?" and "what does it do?"? (both source documentation and [higher level](https://github.com/storj/docs), diagrams?)
 - [x] Does any documentation need updating?
 - [x] Do the database access patterns make sense?

### CODE REVIEW

File: satellite/satellitedb/users.go [hunk 1]
```diff
@@ -54,7 +54,11 @@ func (users *users) Insert(ctx context.Context, user *console.User) (_ *console.
 	}
 
 	optional := dbx.User_Create_Fields{
-		ShortName: dbx.User_ShortName(user.ShortName),
+		ShortName:      dbx.User_ShortName(user.ShortName),
```
Comment (mobyvb) [inline satellite/satellitedb/users.go:57]: For all these nullable fields, you only want to set them if `IsProfessional` is true.
So you can do something like
```
optional := dbx.User_Create_Fields{
...
    ShortName: dbx.User_ShortName_Null()
...
}
if user.IsProfessional {
    optional.ShortName = dbx.User_ShortName(user.ShortName)
    ...
}
```

File: satellite/satellitedb/users.go [hunk 2]
```diff
@@ -163,6 +169,24 @@ func userFromDBX(ctx context.Context, user *dbx.User) (_ *console.User, err erro
 		result.ShortName = *user.ShortName
 	}
 
+	if user.isProfessional {
```
Comment (mobyvb) [inline satellite/satellitedb/users.go:172]: you don't need this check. You should just be able to check `if user.Position != nil` for example. If you follow my other comment, the field will be nil anyway if `isProfessional` is false, so the extra check is unnecessary.

File: satellite/console/users_test.go [hunk 3]
```diff
@@ -18,14 +18,20 @@ import (
...
+	newName           = "newName"
+	newLastName       = "newLastName"
+	newEmail          = "newEmail@mail.test"
+	newPass           = "newPass1234567890123456789012345"
+	position          = "position"
+	compnayName       = "companyName"
+	compnaySize       = 123
+	workingOn         = "workingOn"
+	isProfessional    = true
+	isNotProfessional = false
```
Comment (mobyvb) [inline satellite/console/users_test.go:34]: doesn't look like you need `isNotProfessional`

**Review by mobyvb — COMMENTED** [review]
looks good. Should be fine if you can get the tests to pass

File: satellite/console/users_test.go [hunk 4]
```diff
@@ -128,6 +149,11 @@ func testUsers(ctx context.Context, t *testing.T, repository console.Users, user
 		assert.Equal(t, userByID.PasswordHash, userByEmail.PasswordHash)
 		assert.Equal(t, userByID.PartnerID, userByEmail.PartnerID)
 		assert.Equal(t, userByID.CreatedAt, userByEmail.CreatedAt)
+		assert.Equal(t, userByID.IsProfessional, userByEmail.IsProfessional)
```
Comment (mobyvb) [inline satellite/console/users_test.go:152]: This is only testing that the results of GetByEmail() and Get() match - but we probably also want to compare against the original user. See these lines:
```
assert.Equal(t, name, userByEmail.FullName)
assert.Equal(t, lastName, userByEmail.ShortName)
assert.Equal(t, user.PartnerID, userByEmail.PartnerID)
```
below repository.GetByEmail() and repository.Get()

Maybe we can add some asserts below each one like
```
if user.IsProfessional {
    assert.Equal(t, position, userByEmail.Position)
    ...
} else {
    assert.Equal(t, "", userByEmail.Position)
    ...
}
```

**Review by mobyvb — APPROVED** [review]

**Review by VitaliiShpital — APPROVED** [review]

**Pull request merged**